src/extension.ts

```diff
@@ -188,7 +188,7 @@ export function activate(context: ExtensionContext) {
         window.onDidCloseTerminal(deleteTerminal),
     );
 
-    if (config.get<boolean>('sessionWatcher')) {
+    if (workspace.getConfiguration('r').get<boolean>('sessionWatcher')) {
```

Comment:
The session watcher initialization here only occurs when extension is activated. If user enables it for the first time, it won't work immediately.

Reply:
Thanks for checking! You're right. It looks like the session watcher *would* need `onDidChangeConfiguration` in order for the user to activate it mid-session.

> Instead, we could add something like Restart required to take effect to the description of r.sessionWatcher to make it clear.

I think this is a good approach for now. I will add a commit for this.